frontend/src/features/map_planning/components/timeline/TimelineDatePicker.tsx

```diff
@@ -203,6 +207,51 @@ const TimelineDatePicker = ({ onSelectDate, onLoading, defaultDate }: TimelineDa
     }
   };
 
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    switch (event.key) {
+      case 'ArrowUp':
+        setFocus('up');
+        break;
+      case 'ArrowDown':
+        setFocus('down');
+        break;
+      default:
+        break;
+    }
+  };
+
+  const setFocus = (direction: 'up' | 'down') => {
+    if (direction === 'up') {
+      switch (focusedSlider) {
+        case 'year':
+          setFocusedSlider('day');
+          break;
+        case 'month':
+          setFocusedSlider('year');
+          break;
+        case 'day':
+          setFocusedSlider('month');
+          break;
+        default:
+          break;
+      }
+    } else if (direction === 'down') {
+      switch (focusedSlider) {
+        case 'year':
+          setFocusedSlider('month');
+          break;
+        case 'month':
+          setFocusedSlider('day');
+          break;
+        case 'day':
+          setFocusedSlider('year');
+          break;
+        default:
+          break;
+      }
+    }
+  };
+
   useEffect(() => {
```

Comment:
Can this be broken out into an event handler function?
The state change is somewhere invoked, there should also be the place where the callbacks are called.
Also please justify when using eslint-disable-next-line react-hooks/exhaustive-deps.